eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -61,13 +129,274 @@ protected function configure()
+            ->orderBy('c.modified')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, PDO::PARAM_INT)
+            ->setParameter('since', $since->getTimestamp(), PDO::PARAM_INT);
+
+        return $q->execute();
+    }
+
+    /**
+     * @param mixed $locationId
+     * @param bool $count
+     *
+     * @return \Doctrine\DBAL\Driver\Statement
+     */
+    private function getStatementSubtree($locationId, $count = false)
+    {
+        /**
+         * @var \eZ\Publish\SPI\Persistence\Content\Location\Handler
+         */
+        $locationHandler = $this->getContainer()->get('ezpublish.spi.persistence.location_handler');
+        $location = $locationHandler->load($locationId);
+        $q = $this->connection->createQueryBuilder()
+            ->select($count ? 'count(DISTINCT c.id)' : 'DISTINCT c.id')
+            ->from('ezcontentobject', 'c')
+            ->innerJoin('c', 'ezcontentobject_tree', 't', 't.contentobject_id = c.id')
+            ->where('c.status = :status')
+            ->andWhere('t.path_string LIKE :path')
+            ->orderBy('t.path_string')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, PDO::PARAM_INT)
+            ->setParameter('path', $location->pathString . '%', PDO::PARAM_STR);
+
+        return $q->execute();
+    }
+
+    /**
+     * @param bool $count
+     *
+     * @return \Doctrine\DBAL\Driver\Statement
+     */
+    private function getStatementContentAll($count = false)
+    {
+        $q = $this->connection->createQueryBuilder()
+            ->select($count ? 'count(c.id)' : 'c.id')
+            ->from('ezcontentobject', 'c')
+            ->where('c.status = :status')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, PDO::PARAM_INT);
+
+        return $q->execute();
+    }
+
+    /**
+     * @param \Doctrine\DBAL\Driver\Statement $stmt
+     * @param int $iterationCount
+     *
+     * @return \Generator Return an array of arrays, each array contains content id's of $iterationCount.
+     */
+    private function fetchIteration(Statement $stmt, $iterationCount)
+    {
+        do {
+            $contentIds = [];
+            for ($i = 0; $i < $iterationCount; ++$i) {
+                if ($contentId = $stmt->fetch(PDO::FETCH_COLUMN)) {
+                    $contentIds[] = $contentId;
+                } else {
+                    break;
+                }
+            }
+
+            yield $contentIds;
+        } while (!empty($contentId));
+    }
+
+    /**
+     * @param array $contentIds
+     * @param bool $commit
+     *
+     * @return \Symfony\Component\Process\Process
+     */
+    private static function getPhpProcess(array $contentIds, $commit)
+    {
+        $phpFinder = new PhpExecutableFinder();
+        if (!$phpPath = $phpFinder->find()) {
+            throw new \RuntimeException(
+                'The php executable could not be found, add it to your PATH environment variable and try again'
+            );
+        }
```

Comment:
Didn't notice this before, it could be determined and set in `initialize` instead of on every call to this method, right?

Reply:
Fixed in f57b801 _(but not in `initialize` as this is only needed when running parallel processes)_